app/controllers/predicates/AuthorisedAction.scala

```diff
@@ -127,20 +126,24 @@ class AuthorisedAction @Inject()()(implicit val authConnector: AuthConnector,
     case _: NoActiveSession =>
       logger.info(s"[AuthorisedAction][agentAuthentication] - No active session.")
       unauthorized
-    case _: AuthorisationException =>
-      if (appConfig.emaSupportingAgentsEnabled) {
-        authorised(secondaryAgentPredicate(mtdItId))
-          .retrieve(allEnrolments) {
-            populateAgent(block, mtdItId, _)
-          }.recoverWith {
-          case _ =>
+    case _: AuthorisationException if appConfig.emaSupportingAgentsEnabled =>
+      authorised(secondaryAgentPredicate(mtdItId))
+        .retrieve(allEnrolments) {
+          populateAgent(block, mtdItId, _)
+        }.recoverWith {
+          case _: AuthorisationException =>
             logger.info(s"[AuthorisedAction][agentAuthentication] - Agent does not have secondary delegated authority for Client.")
```

Comment:
i am not sure if this was existing or overlooked, i guess these should be marked as logger.error